demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -291,6 +908,24 @@ static bool downloadS3ObjectFile( const TransportInterface_t * pTransportInterfa
     /* curByte indicates which starting byte we want to download next. */
     size_t curByte = 0;
 
+    /* Variable to store hash digest of payload. */
+    char * pPayloadHashDigest = ( char * ) malloc( SHA256_HASH_DIGEST_LENGTH );
+    /* Variable to store hex encoded hash digest of payload. */
+    char * hexencoded = ( char * ) malloc( SHA256_HEX_ENCODED_DIGEST_LENGTH );
+    /* Store Authorization header value generated using SigV4 library. */
+    char * pSigv4Auth = ( char * ) malloc( AWS_HTTP_AUTH_HEADER_VALUE_LEN );
```

Comment:
Instead of allocating on the heap, it would be better to allocate it locally OR in the global space statically